lib/graphql/batch/loader.rb

```diff
@@ -42,13 +42,14 @@ def current_executor
       end
     end
 
-    attr_accessor :loader_key, :executor
+    attr_accessor :loader_key, :executor, :deferred
```

Comment:
Yes, I agree. I'd like to use `yield` and `yielded`, which is what https://graphql-ruby.org/dataloader/parallelism uses, but that's because it uses fibers. We're conceptually "yielding control" here too.

WDYT @gmac @swalkinshaw @amomchilov ?

Reply:
Since this is an internal implementation detail now, why not just call it `async`? And it's more consistent with the `Async` module naming. 

We could even get rid of this entirely and just filter out loaders that don't include `GraphQL::Batch::Async`? (though maybe that's worse for perf)

Reply:
Can you put a commit up showing what it looks like?